Velaptor/Graphics/RenderMediator.cs

```diff
@@ -4,41 +4,224 @@
+            {
+                continue;
+            }
+
+            this.allLayers.Span[layerIndex] = fontLayer;
+            layerIndex++;
+        }
+
+        for (var i = 0; i < rectItems.Length; i++)
+        {
+            var rectLayer = rectItems.Span[i].Layer;
+            if (this.allLayers.Span.Contains(rectLayer))
+            {
+                continue;
+            }
+
+            this.allLayers.Span[layerIndex] = rectLayer;
+            layerIndex++;
+        }
+
+        for (var i = 0; i < lineItems.Length; i++)
+        {
+            var lineLayer = lineItems.Span[i].Layer;
+            if (this.allLayers.Span.Contains(lineLayer))
+            {
+                continue;
+            }
+
+            this.allLayers.Span[layerIndex] = lineLayer;
+            layerIndex++;
+        }
+
+        this.allLayers.Span.Sort();
+
+        // Render all of the items in a coordinated fashion
+        for (var i = 0; i < this.allLayers.Length; i++)
+        {
+            if (this.allLayers.Span[i] == int.MaxValue)
+            {
+                break;
+            }
+
+            var currentLayer = this.allLayers.Span[i];
+
+            var totalTexturesOnCurrentLayer = textureItems.TotalOnLayer(currentLayer);
+            var totalFontOnCurrentLayer = fontItems.TotalOnLayer(currentLayer);
+            var totalRectsOnCurrentLayer = rectItems.TotalOnLayer(currentLayer);
+            var totalLinesOnCurrentLayer = lineItems.TotalOnLayer(currentLayer);
+
+            if (totalTexturesOnCurrentLayer > 0)
+            {
+                var textureLayerStart = textureItems.FirstLayerIndex(currentLayer);
+
+                this.textureRenderBatchReactable.Push(
+                    textureItems.Slice(textureLayerStart, totalTexturesOnCurrentLayer),
+                    PushNotifications.RenderTexturesId);
+            }
+
+            if (totalFontOnCurrentLayer > 0)
+            {
+                var fontLayerStart = fontItems.FirstLayerIndex(currentLayer);
+
+                this.fontRenderBatchReactable.Push(
+                    fontItems.Slice(fontLayerStart, totalFontOnCurrentLayer),
+                    PushNotifications.RenderFontsId);
+            }
+
+            if (totalRectsOnCurrentLayer > 0)
+            {
+                var rectLayerStart = rectItems.FirstLayerIndex(currentLayer);
+
+                this.rectRenderBatchReactable.Push(
+                    rectItems.Slice(rectLayerStart, totalRectsOnCurrentLayer),
+                    PushNotifications.RenderRectsId);
+            }
+
+            if (totalLinesOnCurrentLayer > 0)
+            {
+                var lineLayerStart = lineItems.FirstLayerIndex(currentLayer);
+
+                this.lineRenderBatchReactable.Push(
+                    lineItems.Slice(lineLayerStart, totalLinesOnCurrentLayer),
+                    PushNotifications.RenderLinesId);
+            }
+
+            // Reset the item back to the default value
```

Comment:
Should this be Reset or Resets?
```suggestion
            // Resets the item back to the default value
```